README.md

```diff
@@ -177,239 +189,175 @@ sergio@sergio:~/plz/examples/pytorch$ plz history | \
+all the time.
 
 ## How does it work?
 
-There is a service called `controller`, and a command-line interface (CLI) that
-performs requests to the controller. The CLI is an executable `plz` accepting
-operation arguments, so that you type `plz run`, `plz stop`, `plz list`, etc.
+Plz consists of a _controller_ service and a _command-line interface_ (CLI) that
+issues requests to the controller. The CLI is a Python executable, `plz`, which
+takes instructions (such as `plz run ...`) as described above.
 
-There are two configurations of the controller that are ready for you to use:
-in one of them your jobs are run locally, while in the other one an aws instance
-is started for each job. (Note: the controller itself can be deployed to the
-cloud --and if you're in a production environment that's the recommended way
-to use it--, but we suggest you try the examples with a controller that runs
-locally first.)
+There are two configurations of the controller that are ready for you to use: in
+one of them your jobs are run locally, while in the other one an AWS instance is
+started for each job. (Note: the controller itself can be deployed to the cloud,
+and if you're in a production environment that's the recommended way to use it,
+but we suggest you try the examples with a controller that runs locally first.)
 
 When you have a directory with source code, you can just add a `plz.config.json`
 file including information such as:
-- The command you want to run
-- The location of your input data
-- Whether you want to request for an instance at fix price, or bid for spot
-  instances, how much money you want to bid, etc.
 
-Then, just typing `plz run` will run the job for you, either locally or in
-aws, depending on the controller you've started.
+- the location of your Plz server,
+- the command you want to run,
+- the location of your input data,
+- whether you want to request an on-demand instance at a fixed price, or bid for
+  spot instances with a ceiling,
+- and much more.
 
+Then, just typing `plz run` will run the job for you, either locally or on AWS,
+depending on the controller you've started.
 
 ## Installation instructions
 
-Chances are you that you have most of the supporting tools already
-installed, as these are broadly used tools. 
+Chances are you that you have most of the supporting tools already installed, as
+these are broadly used tools.
 
-A full list of instructions for Ubuntu is listed below (we've used the tool in
-both Ubuntu and Mac), but in summary you need to: install docker; install
-docker-compose; install the aws CLI, and configure it with your Access key;
-`git clone https://github.com/prodo-ai/plz.git`; install the cli by running 
-`./install_cli` inside the `plz/` directory; run the controller with a script
-we provide.
+1. Install Git, and Python 3. a. On Ubuntu, you can run `sudo apt install git python3 python3-pip`. b. On macOS, install [Homebrew](https://brew.sh/), then
```

Comment:
I think we really, really, need to keep the previous list of instructions for installing (with commands that the user can copy and paste if they have no idea, or just skim through if they have docker, etc.). If there's an independent probability p of the user giving up when needing to click a link, scroll to the installation instructions, etc. then each link increases exponentially the probability of the user giving up :)
If you want a new section for Mac installation: yes, I agree that's great.

Reply:
The problem is, the instructions we had were wrong. The reason the instructions for installing Docker are long and complicated is because there are a lot of edge cases, and by glossing over them, we might end up helping the user break their computer. (For example, they could end up with two conflicting versions of Docker at once.)

So either we reproduce them faithfully, which is pointless, or we just link to them. I am inclined to think that linking to things is more healthy.